JS/Lesson34/onur/my-app/src/components/NewRecipe/index.jsx

```diff
@@ -22,56 +23,195 @@ export const NewRecipe = () => {
 
 	const strCategory = watch('strCategory');
 
-	const onSubmit = (data) => console.log(data);
+
+
+
+	const onSubmit = (data) => {
+
+		if (data.strDrinkAlternate.trim() === '') {
+			data.strDrinkAlternate = null;
+		}
+		if (data.strImageSource.trim() === '') {
+			data.strImageSource = null;
+		}
+		if (data.strCreativeCommonsConfirmed.trim() === '') {
+			data.strCreativeCommonsConfirmed = null;
+		}
+		if (data.dateModified.trim() === '') {
+			data.dateModified = null;
+		}
+
+
+
+
+
+		const ingredientsArray = data.ingredients.split(",").map((ingredient) => ingredient.trim());
+		ingredientsArray.map((ingredient, index) => {
+			for (let i = 1; i <= 20; i++) {
+				if (ingredient.trim().length > 0 && (i <= ingredientsArray.length)) {
+					data[`strIngredient${index + 1}`] = ingredient;
+				} else {
+					data[`strIngredient${i}`] = "";
+				}
+			}
+		});
+		delete data.ingredients;
```

Comment:
I suppose you want to clear input filed by doing `delete data.ingredients;`. It does not work like this. Instead, there's a `reset` function: [reset](https://www.react-hook-form.com/api/useform/reset/).
For reset to work, you must provide default values to `useForm();`.
E.g. `const {reset} = useForm({strMeal: '', strDrinkAlternate:'', ...});

Reply:
Actually, I do not want to clean the entry filed here. After creating values such as Ingredients1, Ingredients2, Ingredients3, a separate string named ingredients was also created in the data. I wanted to have only Ingredients{i} values, so I removed the ingredients string.